turtle_graphics/turtle.js

```diff
@@ -0,0 +1,130 @@
+class Turtle {
+  constructor(initialX, initialY) {
+    this.x = initialX;
+    this.y = initialY;
+    this.path = [[this.x, this.y]];
+    this.direction = "x";
+    this.maxY = -Infinity;
+    this.minY = Infinity;
+    this.maxX = -Infinity;
+    this.minX = Infinity;
```

Comment:
No need to define these as "instance" variables in the constructor. They're only used by the `print` method, and as such should probably stay contained there. I think the reason you moved them here is because you wanted to set them via the `canvasSize` method, but I think even then, you probably want to return these values from that method rather than setting them as instance variables. The only situation in which you'd want to do it this way would be if calculating the `canvasSize` was expensive. Actually, another way you could do this, without needing the `canvasSize`, but preserving the instance variables, would be if you updated the minima and maxima each time you call `forward`.
```javascript
forward(steps) {
  for (...) {
  }
  this.path.push([this.x, this.y]);
  updateBoundaries()
  return this;
}

updateBoundaries() {
  this.minX = Math.min(this.minX, this.x);
  this.minY = Math.min(this.minY, this.y);
  this.maxX = Math.min(this.maxX, this.x);
  this.maxY = Math.min(this.maxY, this.y);
}
```